File: java/abc192/C.java
import java.util.*;

public class C {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();

    int val = n;
    for (int i = 1; i < k + 1; i++) {
      val = func(val);
    }
    System.out.println(val);
  }

  private static int func(int x) {
    String[] s = Integer.toString(x).split("");
    int[] x_arr = Arrays.stream(s).mapToInt(y -> Integer.parseInt(y)).toArray();
    Arrays.sort(x_arr);

    int max = 0;
    int min = 0;
    int idx = 0;
    for (int e : x_arr) {
      max += e * (int) Math.pow(10.0, (double) idx);
      min += e * (int) Math.pow(10.0, (double) x_arr.length - idx - 1);
      idx += 1;
    }
    return max - min;
  }
}

File: rust/abc192/src/c.rs
use proconio::input;
use std::collections::HashMap;
use std::io::{stdout, BufWriter, Write};

pub fn main() {
  let out = stdout();
  let mut out = BufWriter::new(out.lock());

  input! {n: usize, k: usize}
  let val = (0..k).fold(n, |val, _| func(val));

  writeln!(out, "{}", val).unwrap();
}

fn func(v: usize) -> usize {
  let mut vect: Vec<usize> = v
    .to_string()
    .chars()
    .map(|c| c.to_string().parse::<usize>().unwrap())
    .collect();
  vect.sort();

  let (max, min) = vect.iter().enumerate().fold((0, 0), |acc, (idx, e)| {
    (
      acc.0 + e * 10usize.pow(idx as u32),
      acc.1 + e * 10usize.pow((vect.len() - idx - 1) as u32),
    )
  });

  max - min
}
